tests/integration/attachment_test.go

```diff
@@ -41,7 +41,15 @@ func createAttachment(t *testing.T, session *TestSession, repoURL, filename stri
 	err = writer.Close()
 	assert.NoError(t, err)
 
-	csrf := GetCSRF(t, session, repoURL)
+	var csrf string
+	// FIXME: It's quite hacky to determine if it's a logged in user or not and use the right URL to get the CSRF token
+	if expectedStatus == http.StatusSeeOther {
+		// the session is not logged in
+		csrf = GetCSRF(t, session, "/user/login")
+	} else {
+		// the session is logged in
+		csrf = GetCSRF(t, session, repoURL)
+	}
```

Comment:
the CSRF token could be prepared by the caller of createAttachment, then this "FIXME" could be perfectly resolved.

eg: 

```
createAttachment(t, session, GetCSRF(repoURL) ...)
createAttachment(t, session, GetCSRF("/user/login") ...)
```

Reply:
Good idea, 58bc78645f62edb8ed7f7ba14d707b98d56ddf5b